src/main/java/org/dmfs/jems/charsequence/elementary/Ascii.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.charsequence.elementary;
+
+import org.dmfs.jems.single.Single;
+import org.dmfs.jems.single.elementary.Frozen;
+
+import java.util.Locale;
+
+
+/**
+ * A {@link CharSequence} adapter to ASCII encoded byte arrays.
+ *
+ * @author Marten Gajda
+ */
+public final class Ascii implements CharSequence
+{
+    private final byte[] mData;
+    private final int mStart;
+    private final int mEnd;
+    private final Single<String> mToString;
+
+
+    public Ascii(byte... data)
+    {
+        this(data, 0, data.length);
+    }
+
+
+    public Ascii(byte[] data, int start, int end)
+    {
+        mData = data;
+        mStart = start;
+        mEnd = end;
+        mToString = new Frozen<>(new Single<String>()
+        {
+            @Override
+            public String value()
+            {
+                StringBuilder stringBuilder = new StringBuilder(mEnd - mStart);
```

Comment:
Does this `StringBuilder` composition have benefit over `new String(data, "UTF-8")`?

Reply:
We can avoid the rather expensive character encoding process by just "casting" the byte to `UTF-16`, which is always valid for ASCII encoded characters. So even though the `StringBuilder` requires one more copy operation, I believe it's most likely faster than the `String` ctor.